src/decompose_mesh/rules.mk

```diff
@@ -106,7 +130,10 @@ endif
 
 $O/decompose_mesh.dec.o: $O/part_decompose_mesh.dec.o $O/fault_scotch.dec.o ${SCOTCH_DIR}/include/scotchf.h $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
 $O/program_decompose_mesh.dec.o: $O/decompose_mesh.dec.o $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
-
+$O/program_decompose_mesh_mpi.mpidec.o: $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
```

Comment:
There are a few other dependencies missing, I think. From my look through:
*    decompose_mesh/module_mesh.f90
       * fault_scotch
*    decompose_mesh/module_partition.f90
       * module_qsort
*    decompose_mesh/program_decompose_mesh_mpi.f90
       * module_mesh
       * module_database
       * module_partition

Also, it is not necessary to specify `$O/shared_par.shared_module.o` here; it is already a dependency from the `$O/%.dec.o` rule below.